pkg/registry/softwarecomposition/networkneighbors/strategy.go

```diff
@@ -0,0 +1,121 @@
+	return generic.ObjectMetaFieldsSet(&obj.ObjectMeta, true)
+}
+
+type networkNeighborsStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+func (networkNeighborsStrategy) NamespaceScoped() bool {
+	return true
+}
+
+func (networkNeighborsStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
+}
+
+func (networkNeighborsStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+}
+
+func (networkNeighborsStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
+	networkNeighbors := obj.(*softwarecomposition.NetworkNeighbors)
+
+	return validatePorts(networkNeighbors)
+}
+
+// WarningsOnCreate returns warnings for the creation of the given object.
+func (networkNeighborsStrategy) WarningsOnCreate(ctx context.Context, obj runtime.Object) []string {
+	return nil
+}
+
+func (networkNeighborsStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+func (networkNeighborsStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+func (networkNeighborsStrategy) Canonicalize(obj runtime.Object) {
+}
+
+func (networkNeighborsStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	networkNeighbors := obj.(*softwarecomposition.NetworkNeighbors)
+
+	return validatePorts(networkNeighbors)
+}
+
+// WarningsOnUpdate returns warnings for the given update.
+func (networkNeighborsStrategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
+	return nil
+}
+
+func validatePorts(networkNeighbors *softwarecomposition.NetworkNeighbors) field.ErrorList {
+	for _, ingress := range networkNeighbors.Spec.Ingress {
+		for _, networkPort := range ingress.Ports {
+			if *networkPort.Port < 0 || *networkPort.Port > 65535 {
+				return field.ErrorList{field.Invalid(field.NewPath("spec").Child("ingress").Child("ports"), *networkPort.Port, "port must be in range 0-65535")}
+			}
+
+			expectedPortName := fmt.Sprintf("%s-%d", networkPort.Protocol, *networkPort.Port)
+			if networkPort.Name != expectedPortName {
+				return field.ErrorList{field.Invalid(field.NewPath("spec").Child("ingress").Child("ports"), *networkPort.Port, fmt.Sprintf("port name must be in format {protocol}-{port}, expected name: %s", expectedPortName))}
+			}
+		}
+	}
+
+	for _, egress := range networkNeighbors.Spec.Egress {
+		for _, networkPort := range egress.Ports {
+			if *networkPort.Port < 0 || *networkPort.Port > 65535 {
+				return field.ErrorList{field.Invalid(field.NewPath("spec").Child("egress").Child("ports"), *networkPort.Port, "port must be in range 0-65535")}
+			}
+
+			expectedPortName := fmt.Sprintf("%s-%d", networkPort.Protocol, *networkPort.Port)
+			if networkPort.Name != expectedPortName {
+				return field.ErrorList{field.Invalid(field.NewPath("spec").Child("egress").Child("ports"), *networkPort.Port, fmt.Sprintf("port name must be in format {protocol}-{port}, expected name: %s", expectedPortName))}
+			}
+		}
+	}
```

Comment:
suggestion: ...and once you split the validators as above, you wouldn’t need to duplicate the code